Technico/Services/UserService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Technico.Models;
+using Technico.Repositories;
+using Technico.Dtos;
+
+namespace Technico.Services;
+
+public class UserService
+{
+    private readonly UserRepository _userRepository;
+
+    public UserService(UserRepository userRepository)
+    {
+        _userRepository = userRepository;
+    }
+
+    public async Task<UserResponseDTO?> CreateAsync(UserRequestDTO createDto)
+    {
+        var users = await _userRepository.GetAllAsync();
+
+        bool VATexists = users.Any(u => u.VATNumber == createDto.VATNumber);
```

Comment:
The CreateAsync method checks for existing VATNumber and Email using LINQ queries on all users. This could be encapsulated into the repository layer with specific query methods (e.g., ExistsByVATNumberAsync or ExistsByEmailAsync) for better separation of concerns and potential performance improvements.